nestjs/src/courses/course-students/course-students.controller.ts

```diff
@@ -0,0 +1,46 @@
+import { Controller, Get, NotFoundException, Param, Req, UseGuards } from '@nestjs/common';
+import { ApiBadRequestResponse, ApiForbiddenResponse, ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { CurrentRequest, DefaultGuard } from '../../auth';
+import { StudentSummaryDto } from './dto/student-summary.dto';
+import { CourseStudentsService } from './course-students.service';
+
+@Controller('courses/:courseId/students')
+@ApiTags('students')
+@UseGuards(DefaultGuard)
+export class CourseStudentsController {
+  constructor(private courseStudentService: CourseStudentsService) {}
+
+  @Get(':githubId/summary')
+  @ApiForbiddenResponse()
+  @ApiBadRequestResponse()
+  @ApiOkResponse({
+    type: StudentSummaryDto,
+  })
+  @ApiOperation({ operationId: 'getStudentSummary' })
+  public async getStudentSummary(
+    @Param('courseId') courseId: number,
+    @Param('githubId') githubId: string,
+    @Req() req: CurrentRequest,
+  ) {
+    const studentGithubId = githubId === 'me' ? req.user.githubId : githubId;
```

Comment:
To be honest, I don't like this usage of `githubId`, such profile exist in github (https://github.com/me). Let's replace it with separated param that explicitly means that we should use user's github id